code/modules/mod/modules/modules_general.dm

```diff
@@ -471,6 +471,13 @@
 	incompatible_modules = list(/obj/item/mod/module/plasma_stabilizer)
 	overlay_state_inactive = "module_plasma"
 
+/obj/item/mod/module/plasma_stabilizer/add_module_overlay()
+	if(locate(/obj/item/mod/module/infiltrator) in mod.modules)
+		overlay_state_inactive = null
+	else
+		overlay_state_inactive = initial(overlay_state_inactive)
+	..()
```

Comment:
this is a bad way to do this.
i dont see why you would wanna set the overlay state here
just do an early return in generate_worn_overlay of this module

Reply:
Done, sorry for that, I'm not very familiar with the code